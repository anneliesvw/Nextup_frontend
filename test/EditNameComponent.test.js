import Vuex from 'vuex';
import { shallow, createLocalVue } from 'vue-test-utils';
import EditNameComp from '../src/components/users/dialogs/EditName.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('EditNameComp', () => {
  let store;
  let actions;
  beforeEach(() => {
    actions = {
      updateUser: jest.fn(),
    };
    store = new Vuex.Store({
      actions,
    });
  });

  // faalt?

  it('dispatches an action when a button is clicked', () => {
    const wrapper = shallow(EditNameComp, {
      store,
      localVue,
      data: {
        userInfo: {
          person: { firstName: 'Jef', lastName: 'Test', location: { name: 'TestLocation' } },
          username: 'TestUser',
        },
      },
    });
    wrapper.find('.update-user-btn').trigger('click');
    expect(actions.updateUser.mock.calls).toHaveLength(1);
  });
});
