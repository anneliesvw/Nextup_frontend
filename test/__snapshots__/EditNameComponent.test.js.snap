// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNameComp should match snapshot 1`] = `
<el-dialog
  title="Edit name"
>
  <div
    class="edit-name-form"
  >
    <el-form
      class="edit-name-form"
      label-position="top"
    >
      <el-form-item
        label="Firstname"
      >
        <el-input
          class="edit-name-input"
        />
      </el-form-item>
       
      <el-form-item
        label="Lastname"
      >
        <el-input
          class="edit-name-input"
        />
      </el-form-item>
       
      <el-form-item
        label="Name"
      >
        <el-input
          class="edit-name-input"
          disabled="disabled"
        />
      </el-form-item>
    </el-form>
  </div>
   
  <span
    class="dialog-footer"
    slot="footer"
  >
    <el-button>
      Cancel
    </el-button>
     
    <el-button
      class="update-user-btn"
      type="primary"
    >
      Save
    </el-button>
  </span>
</el-dialog>
`;
